GTViewController/Classes/Custom Classes/UISnuffleButton.m

```diff
@@ -445,15 +454,19 @@ - (void)touchesEnded:(NSSet *)touches withEvent:(UIEvent *)event {
 				
 			}
 			
-            [self.tapEvents enumerateObjectsUsingBlock:^(NSString *eventName, NSUInteger idx, BOOL *stop) {
-                if (eventName) {
-                    NSString *trimmedEventName = [eventName stringByTrimmingCharactersInSet:[NSCharacterSet whitespaceCharacterSet]];
-                    
-                    [[NSNotificationCenter defaultCenter] postNotificationName:UISnuffleButtonNotificationButtonTapEvent
-                                                                        object:self
-                                                                      userInfo:@{UISnuffleButtonNotificationButtonTapEventKeyEventName: trimmedEventName}];
-                }
-            }];
+            if (self.forceValidation && self.tapDelegate && [self.tapDelegate respondsToSelector:@selector(didReceiveTapOnPositiveButton:)]) {
```

Comment:
If you want to keep this pattern, which I don't recommend. You need to implement this method on GTPage so that if a positive button appears on the page or in a panel it still fires the events. You will also need to deal with the fact that the didReceiveTapOnPositiveButton is only called if forceValidation is set, which is not necessarily going to be true if it appears elsewhere in the xml.